src/runtime/opengl.cpp

```diff
@@ -821,8 +821,8 @@ WEAK int halide_opengl_device_malloc(void *user_context, buffer_t *buf) {
 
         // Set parameters for this texture: no interpolation and clamp to edges.
         global_state.BindTexture(GL_TEXTURE_2D, tex);
-        global_state.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
-        global_state.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+        global_state.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
```

Comment:
One of the use cases for this PR is to enable apps to use interpolated texture sampling by making "extern" calls to texture2D. This change enables this sampling on textures created by the Halide runtime. Halide ordinarily maps integer Func coordinates to texel center coordinates- so there is no change in behavior for the ordinary glsl_texture_load intrinsic case. I'll add this description to the comment.